src/core/border_router/routing_manager.cpp

```diff
@@ -2170,13 +2198,34 @@ Error RoutingManager::LocalOnLinkPrefix::Advertise(void)
 
     Error error = kErrorNone;
 
-    VerifyOrExit(mState != kAdvertising);
+    switch (mState)
+    {
+    case kIdle:
+    case kDeprecating:
+        break;
+
+    case kPublishing:
```

Comment:
Not necessary to cover in this PR.

If we once failed to publish the external route for the on-link prefix, we will stay in the `kPublishing` state indefinitely. Maybe we can retry a few times in this case.

Reply:
Good point. I think the only way for `PublishExtenralRoute()` to fail is if we run out of entries in the `NetData::Publisher` itself (which uses a fixed array/pool). We have a large pool so unlikely for it to fail but possible.

One idea which I think may be good for this is to change the `Publisher` to keep certain entries in pool reserved for `RoutingManager` use only. We can know max number of entries we can publish from `RoutingManager` (derive it from max number of entries in `DiscoveredPrefixTable` and local on-link prefixes). This way, we can make sure that we would not fail in `PublishExtenralRoute()` and/or we can assert if we do.

I will try to add this as an enhancement later.

Reply:
- Added https://github.com/openthread/openthread/pull/8148.